### ISSUE TITLE
Bug: dvc remote remove leads to Initialization error / config file error.

### ISSUE THREAD

**Issue by tdeboissiere** [iterative/dvc#1075]
**Setup**

- dvc version 0.18.9, installed with pip, python Anaconda 3.6.4, Ubuntu 16.04

**Repro**

```bash
set -e
set -x
rm -rf foo
mkdir -p foo && cd foo
git init && dvc init
echo bar > bar.txt
dvc remote add -d dummy s3://dummy
dvc remote remove dummy
dvc add bar.txt
```

```bash
+ rm -rf foo
+ mkdir -p foo
+ cd foo
+ git init
Initialized empty Git repository in /home/tmain/foo/.git/
+ dvc init
Adding '.dvc/state' to '.dvc/.gitignore'.
Adding '.dvc/state.lock' to '.dvc/.gitignore'.
Adding '.dvc/link.state' to '.dvc/.gitignore'.
Adding '.dvc/link.state.lock' to '.dvc/.gitignore'.
Adding '.dvc/lock' to '.dvc/.gitignore'.
Adding '.dvc/config.local' to '.dvc/.gitignore'.
Adding '.dvc/updater' to '.dvc/.gitignore'.
Adding '.dvc/cache' to '.dvc/.gitignore'.
Checking for updates...
+ echo bar
+ dvc remote add -d dummy s3://dummy
Setting 'dummy' as a default remote.
+ dvc remote remove dummy
+ dvc add bar.txt
Error: Initialization error: Config file error: Can't find remote section 'remote "dummy"' in config

```

**Comment by shcheklein** [comment]
Hi @tdeboissiere, this is definitely a bug. I think this is happening because the remote you removed is assigned as a default one in the `core` config section:

```
['remote "myremote"']
url = /tmp/dvc-storage
[core]
remote = myremote
```

We'll fix this asap, meanwhile, I would try

```
dvc config -u core.remote
```

as a temporary workaround.

Thank you, for the report!

**Comment by efiop** [comment]
Hi @tdeboissiere !

Thanks for the feedback! The patch is merged and released under 0.18.10. Please feel free to upgrade.

Thanks,
Ruslan

**Comment by tdeboissiere** [comment]
Awesome, thanks !
